code/workspaces/client-api/src/dataaccess/login/rstudioTokenService.js

```diff
@@ -47,17 +48,25 @@ const login = (notebook, credentials) => (response) => {
 };
 
 function getRStudioCertificate(notebook) {
-  logger.debug(`Requesting RStudio certificate from ${notebook.internalEndpoint}/auth-public-key`);
-  return axios.get(`${notebook.internalEndpoint}/auth-public-key`)
-    .catch((error) => {
-      throw new Error(`Unable to retrieve RStudio certificate: ${error.message}`);
-    });
+  const notebookUrl = getCorrectAccessUrl(notebook);
```

Comment:
Are there not equivalent changes for Zeppelin and Minio?

Reply:
Good spot! I wasn't that far through the changes from Vault so forgot about these. Will add the necessary updates.